src/Optical/Paraxial.jl

```diff
@@ -17,11 +17,11 @@ ParaxialLensConvexPoly(focaldistance, local_frame, local_polygon_points, local_c
 ```
 """
 struct ParaxialLens{T} <: Surface{T}
-    shape::Union{Rectangle{T},Ellipse{T},Hexagon{T},ConvexPolygon{N, T} where {N}} 
+    shape::PlanarShape{T}
```

Comment:
Using an abstract type here might mean that the type of shape can't be resolved when tracing - you could parameterise ParaxialLens with the shape type or dimensionality as well as type maybe?